### ISSUE TITLE
[v3.0.0-beta.10] GoutteWrapper request incompatible declaration

### ISSUE THREAD

**Issue by WillyReyno** [jikan-me/jikan#379]
Hi!

I'm getting the following error after upgrading PHP to 8.0 as well as all my dependencies: 

```
Declaration of Jikan\Goutte\GoutteWrapper::request(string $method, string $uri, array $parameters = [], array $files = [], array $server = [], ?string $content = null, bool $changeHistory = true) must be compatible with Symfony\Component\BrowserKit\AbstractBrowser::request(string $method, string $uri, array $parameters = [], array $files = [], array $server = [], ?string $content = null, bool $changeHistory = true): Symfony\Component\DomCrawler\Crawler
```

It looks like a `: \Symfony\Component\DomCrawler\Crawler` return type is missing on the `Jikan\Goutte\GoutteWrapper::request()` method. 

I tried to add it, but I get another error from `jikan-me/jikan/src/Model/Anime/Anime.php` on line 223

```
Call to undefined method Symfony\Component\DomCrawler\Crawler::parents()
```

**Comment by irfan-dahir** [comment]
Hey, sorry for the late response. I'm not sure what's causing this issue. Can you re-install the dependencies without upgrading them and see if that fixes it for you?

**Comment by WillyReyno** [comment]
Just tried reinstalling and it still doesn't work unfortunately

**Comment by irfan-dahir** [comment]
@WillyReyno Found the issue. 
It was caused by a new [Goutte release](https://github.com/FriendsOfPHP/Goutte/releases/tag/v4.0.2) `4.0.2` which is causing a breaking change.

Should be fixed by setting goutte version in composer.json to: `"fabpot/goutte": "4.0.1"`

Then remove the `vendor` folder and `composer.lock` file and then `composer install`

Let me know if that works for you - meanwhile, I'll create a release for this.

**Comment by irfan-dahir** [comment]
I've released [v3.0.0-beta.11](https://github.com/jikan-me/jikan/releases/tag/v3.0.0-beta.11) which locks Goutte to `4.0.1`

```
Call to undefined method Symfony\Component\DomCrawler\Crawler::parents()
```
This method does not seem to exist at all in Goutte `4.0.2`. It's used in about 80 occurrences for parsing data so I'll need to find a work-around.

**Comment by WillyReyno** [comment]
Can confirm new release fixed it, thanks!

Should I close this issue or do you prefer leaving it open for you to fix compatibility with Goutte `4.0.2`?

**Comment by irfan-dahir** [comment]
Goutte `4.0.2` uses the next major version of the Symfony Crawler dependencies (https://github.com/FriendsOfPHP/Goutte/commit/f51940fbe0db060bc4fc0b3f1d19bc4ff3054b17) which potentially means a refactoring of many areas plus `4.0.1` works without issue on PHP 8 or 8.1.

So I think we're fine right now and there's no need to upgrade. I'll go ahead and close this.
